File: frontend/lib/features/ai/domain/entities/ai_chat_response.dart
import 'package:equatable/equatable.dart';

class AiChatResponse extends Equatable {
  final String messageId;
  final String conversationId;
  final bool emergency;
  final String reply;
  final String disclaimer;
  final String model;
  final String promptVersion;
  final String createdAt;

  const AiChatResponse({
    required this.messageId,
    required this.conversationId,
    required this.emergency,
    required this.reply,
    required this.disclaimer,
    required this.model,
    required this.promptVersion,
    required this.createdAt,
  });

  factory AiChatResponse.fromJson(Map<String, dynamic> json) {
    return AiChatResponse(
      messageId: json['messageId']?.toString() ?? '',
      conversationId: json['conversationId']?.toString() ?? '',
      emergency: json['emergency'] as bool? ?? false,
      reply: json['reply']?.toString() ?? '',
      disclaimer: json['disclaimer']?.toString() ?? '',
      model: json['model']?.toString() ?? '',
      promptVersion: json['promptVersion']?.toString() ?? '',
      createdAt: json['createdAt']?.toString() ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'messageId': messageId,
      'conversationId': conversationId,
      'emergency': emergency,
      'reply': reply,
      'disclaimer': disclaimer,
      'model': model,
      'promptVersion': promptVersion,
      'createdAt': createdAt,
    };
  }

  @override
  List<Object?> get props => [
        messageId,
        conversationId,
        emergency,
        reply,
        disclaimer,
        model,
        promptVersion,
        createdAt,
      ];
}

File: backend/src/main/java/com/vettrack/api/ai/dto/AiChatResponse.java
package com.vettrack.api.ai.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AiChatResponse {
    private UUID messageId;
    private UUID conversationId;
    private boolean emergency;
    private String reply;
    private String disclaimer;
    private String model;
    private String promptVersion;
    private OffsetDateTime createdAt;

    public static AiChatResponseBuilder builder() {
        return new AiChatResponseBuilder();
    }

    public static class AiChatResponseBuilder {
        private UUID messageId;
        private UUID conversationId;
        private boolean emergency;
        private String reply;
        private String disclaimer;
        private String model;
        private String promptVersion;
        private OffsetDateTime createdAt;

        public AiChatResponseBuilder messageId(UUID messageId) { this.messageId = messageId; return this; }
        public AiChatResponseBuilder conversationId(UUID conversationId) { this.conversationId = conversationId; return this; }
        public AiChatResponseBuilder emergency(boolean emergency) { this.emergency = emergency; return this; }
        public AiChatResponseBuilder reply(String reply) { this.reply = reply; return this; }
        public AiChatResponseBuilder disclaimer(String disclaimer) { this.disclaimer = disclaimer; return this; }
        public AiChatResponseBuilder model(String model) { this.model = model; return this; }
        public AiChatResponseBuilder promptVersion(String promptVersion) { this.promptVersion = promptVersion; return this; }
        public AiChatResponseBuilder createdAt(OffsetDateTime createdAt) { this.createdAt = createdAt; return this; }

        public AiChatResponse build() {
            AiChatResponse r = new AiChatResponse();
            r.messageId = this.messageId;
            r.conversationId = this.conversationId;
            r.emergency = this.emergency;
            r.reply = this.reply;
            r.disclaimer = this.disclaimer;
            r.model = this.model;
            r.promptVersion = this.promptVersion;
            r.createdAt = this.createdAt;
            return r;
        }
    }

    public UUID getMessageId() { return messageId; }
    public void setMessageId(UUID messageId) { this.messageId = messageId; }
    public UUID getConversationId() { return conversationId; }
    public void setConversationId(UUID conversationId) { this.conversationId = conversationId; }
    public boolean isEmergency() { return emergency; }
    public void setEmergency(boolean emergency) { this.emergency = emergency; }
    public String getReply() { return reply; }
    public void setReply(String reply) { this.reply = reply; }
    public String getDisclaimer() { return disclaimer; }
    public void setDisclaimer(String disclaimer) { this.disclaimer = disclaimer; }
    public String getModel() { return model; }
    public void setModel(String model) { this.model = model; }
    public String getPromptVersion() { return promptVersion; }
    public void setPromptVersion(String promptVersion) { this.promptVersion = promptVersion; }
    public OffsetDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(OffsetDateTime createdAt) { this.createdAt = createdAt; }
}
